src/app/signup/signup.component.ts

```diff
@@ -24,72 +26,112 @@ export class SignupComponent implements OnInit {
  
  captcha?: string;
  errormsg: string;
- declarativeFormCaptchaValue:any;
+ window: any;
+ rec_response:any;
+ grecaptcha:any;
+ score:any;
+
+
+  public recentToken: string = '';
+  public readonly executionLog: OnExecuteData[] = [];
+
+  private allExecutionsSubscription: Subscription;
+  private singleExecutionSubscription: Subscription;
 
   
   signupForm: FormGroup;
   isSubmitted:boolean=false;
   constructor( public authService: AuthService,
   	           private fb: FormBuilder, 
   	           private snackbar: MatSnackBar,
+               private recaptchaV3Service: ReCaptchaV3Service
   	           )
                {
   	               this.signupForm = this.fb.group({
                    email: new FormControl('',[Validators.compose([Validators.required,Validators.email,Validators.pattern(/^[_a-z0-9]+(\.[_a-z0-9]+)*@[a-z0-9-]+(\.[a-z0-9-]+)*(\.[a-z]{2,4})$/)]) ]),
                    //captcha: new FormControl('',[Validators.required]),
-                   user_id: new FormControl('',[Validators.required]),                   
+                   username: new FormControl('',[Validators.required]),                   
                    password: ['', [Validators.required,Validators.minLength(6),PasswordValidator.cannotContainSpace]],
                    phone: new FormControl(''),
                     });
 
 
                      
   	            }
-  
+
+
+   
   ngOnInit() {
+    this.loadExternalScript('https://www.google.com/recaptcha/api.js?render=6LdoiLwZAAAAANJ-MV-ZORWzs8IwU1IjDPJcXnvn');
   
   	
   }
+  public loadExternalScript(url: string) {
+  const body = <HTMLDivElement> document.body;
+  const script = document.createElement('script');
+  script.innerHTML = '';
+  script.src = url;
+  script.async = true;
+  script.defer = true;
+  body.appendChild(script);
+}
+
+  
+
   /*
  * This function is used for signup
- * @params(email:string,password:string,user id:string,phone number:string)
- * As  no live api url found we are setting token static value  
+ * @params(email:string,password:string,user name:string,phone number:string)   
  */
-
   signupFormSubmit(){
-      this.isSubmitted = true;
-       const response = grecaptcha.getResponse();
-      console.log("forms"+response);
-      if (response.length === 0) {
-          console.log("not");
-          this.errormsg = 'Recaptcha not verified. Please try again!';
-           this.snackbar.open('Recaptcha not verified. Please try again!','OK',{
-                verticalPosition: 'top',
-                horizontalPosition:'right'
-              });
-          return;
-        }
+      this.isSubmitted = true;         
+      
+      
       if(this.signupForm.valid)
       {
-      let saveData = this.signupForm.value;   
-       
+       let saveData = this.signupForm.value;   
+          grecaptcha.ready(() => {
+              grecaptcha.execute('6LdoiLwZAAAAANJ-MV-ZORWzs8IwU1IjDPJcXnvn', { action: 'cta_signup' }).then((token) => {
```

Comment:
@Bithika-Redapple The action can be a Constant, and as mentioned before the secret can be a constant someplace. Hopefully in  some property or configuration file where it is less obvious where it is at.